past_data_analysis.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+
+from typing import List
+from afdd.utils import * 
+from afdd.models import *
+import psycopg
+import os
+import json
+
+def analyze_past_data(conn: Connection, start_time: str, end_time: str, rule_id: int):
+  """
+  Creates a json file with all anomalies that happened between start and end time for a specified rule.
+  """
+  graph = load_graph(devices="kaiterra_example.ttl")
+  
+  # get full rule from rules table in postgres by rule_id
+  query = f"""SELECT rule_id, name, sensor_type, description, condition
+  FROM rules
+  WHERE rule_id={rule_id}"""
+
+  with conn.cursor() as cur:
+    cur.execute(query)
+    result = cur.fetchall()
+    result = result[0]
+    print(result)

```

Comment:
For queries like this its good practice to use parameterized queries with "%s" instead of string interpolation like python f"" strings. This prevents against SQL injection attacks. The %s ensures the data is treated as a value instead of apart of the sql command.

It's not really a big deal here, but just good practice to follow.

The query would like more like:

```
" where rule_id = %s"

cur.execute(query, (ruleid))
```